examples/self_sustaining_parallel.py

```diff
@@ -0,0 +1,205 @@
+"""
+===========================================================
+Running a self-sustained flow of parallel tasks with joblib
+===========================================================
+
+This example introduces a recipe for implementing a feedback loop where the
+output generator of joblib.Parallel is used to fuel the input generator with
+new tasks, based on the value that are returned by the tasks that have been
+already completed, until some stopping criterion is met.
```

Comment:
Note that in this example the stopping criterion must be implemented within the function that generate new tasks from results, and the first time it is triggered by a task, it will stop the generation of new tasks for the current callback threads and all future callback threads, so the remaining compute left will only consist in completing pending tasks and retrieving the outputs. It is probably enough to cover most usecases I can think of.

It could be extended to an implementation where callbacks can arbitrarily choose to not create a new task, independently from other callbacks, or create several tasks, thus resulting in a flow of task of variable intensity during the compute, but implementing such agency either requires deeper changes to joblib.Parallel (at least adding one additional lock and decouple output retrieval and task submission steps), either hacky workarounds (submitting dummy tasks for the sole purpose of breaking the locks).